src/com/untamedears/realisticbiomes/persist/WorldID.java

```diff
@@ -19,27 +24,53 @@ public static void init (Plugin plugin) {
 		
 		World end = plugin.getServer().getWorld("world_the_end");
 		WorldID.endID = end == null ? null : end.getUID();
+		
+		for (World world: plugin.getServer().getWorlds()) {
+			if (!world.getName().equals("world")
+					&& !world.getName().equals("world_nether")
+					&& !world.getName().equals("world_the_end")) {
+				if (otherWorlds == null) {
+					RealisticBiomes.doLog(Level.WARNING, "There are worlds that don't have default names (world, world_nether, world_the_end), experimental feature");
+					otherWorlds = new HashMap<UUID, Integer>();
+				}
+				otherWorlds.put(world.getUID(), world.getUID().hashCode());
+			}
+		}
 	}
 	
 	public static UUID getMCID(int id) {
-		if (id == 0)
+		if (id == 0) {
 			return overworldID;
-		else if (id == 1)
+		} else if (id == 1) {
 			return netherID;
-		else if (id == 2)
+		} else if (id == 2) {
 			return endID;
-		else
+		} else {
+			for (UUID uid: otherWorlds.keySet()) {
+				if (otherWorlds.get(uid).intValue() == id) {
+					return uid;
+				}
+			}
+			RealisticBiomes.doLog(Level.SEVERE, "Cannot get UUID for world with id: " + id);			
 			return null;
+		}
 	}
 	
 	public static int getPID(UUID id) {
-		if (id == overworldID)
+		if (id == overworldID) {
 			return 0;
-		else if (id == netherID)
+		} else if (id == netherID) {
 			return 1;
-		else if (id == endID)
+		} else if (id == endID) {
 			return 2;
-		else
-			return -1;		
+		} else {
+			for (UUID uid: otherWorlds.keySet()) {

```

Comment:
UUID has a full featured comparable, and you've got it in a hash, don't waste that ln(n) lookup.

<code>
if (otherWorlds.containsKey(uid)) {
    return otherWorlds.get(uid).intValue();
} else {
    RealisticBiomes.doLog(Level.SEVERE, "Cannot get world id for UUID: " + id);
    return -1;
}
</code>

etc.